llama_index/schema.py

```diff
@@ -18,7 +18,11 @@ def _validate_is_flat_dict(metadata_dict: dict) -> None:
         if not isinstance(key, str):
             raise ValueError("Metadata key must be str!")
         if not isinstance(val, (str, int, float)):
```

Comment:
I'm undecided, but maybe this check could be something like `isinstance(val, (str, int, float, type(None)))`

Reply:
Did the type(None) implementation, less impactful and more elegant, again I would like to avoid too much change XD